packages/forklift-console-plugin/src/modules/Providers/utils/components/CertificateUpload/FetchCertificateModal.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { FC, useState } from 'react';
+import { calculateThumbprint, useTlsCertificate } from 'src/modules/Providers/hooks';
+import { useModal } from 'src/modules/Providers/modals';
+import { useForkliftTranslation } from 'src/utils/i18n';
+
+import { Loading } from '@kubev2v/common';
+import { Alert, Button, Modal, ModalVariant } from '@patternfly/react-core';
+
+import { VerifyCertificate } from './VerifyCertificate';
+
+export const FetchCertificateModal: FC<{
+  url: string;
+  existingCert: string;
```

Comment:
what is `existingCert` used for ?

you have `useTlsCertificate`   as the source of the data, when do you use `existingCert` ?

Reply:
This is the certificate already visible on the screen (coming directly from the secret).

Reply:
but you don't care about the current value, you will override it, if it changed or not.